dbt/parser/schemas.py

```diff
@@ -61,186 +62,386 @@ def as_kwarg(key, value):
-        to_return = {}
+    def find_schema_yml(cls, package_name, root_dir, relative_dirs):
+        """This is common to both v1 and v2 - look through the relative_dirs
+        under root_dir for .yml files yield pairs of filepath and loaded yaml
+        contents.
+        """
+        extension = "[!.#~]*.yml"
 
-        for test in tests:
-            raw_yml = test.get('raw_yml')
-            test_name = "{}:{}".format(test.get('package_name'),
-                                       test.get('path'))
+        file_matches = dbt.clients.system.find_matching(
+            root_dir,
+            relative_dirs,
+            extension)
+
+        for file_match in file_matches:
+            file_contents = dbt.clients.system.load_file_contents(
+                file_match.get('absolute_path'), strip=False)
+            test_path = file_match.get('relative_path', '')
+
+            original_file_path = os.path.join(file_match.get('searched_path'),
+                                              test_path)
 
             try:
-                test_yml = dbt.clients.yaml_helper.load_yaml_text(raw_yml)
+                test_yml = dbt.clients.yaml_helper.load_yaml_text(
+                    file_contents
+                )
             except dbt.exceptions.ValidationException as e:
                 test_yml = None
-                logger.info("Error reading {} - Skipping\n{}".format(
-                            test_name, e))
+                logger.info("Error reading {}:{} - Skipping\n{}".format(
+                            package_name, test_path, e))
 
             if test_yml is None:
                 continue
 
-            no_tests_warning = ("* WARNING: No constraints found for model"
-                                " '{}' in file {}\n")
-            for model_name, test_spec in test_yml.items():
-                if test_spec is None or test_spec.get('constraints') is None:
-                    test_path = test.get('original_file_path', '<unknown>')
-                    logger.warning(no_tests_warning.format(model_name,
-                                   test_path))
+            yield original_file_path, test_yml
+
+    @classmethod
+    def parse_v1_test_yml(cls, original_file_path, test_yml, package_name,
+                          root_project, all_projects, root_dir, macros=None):
+        """Parse v1 yml contents, yielding parsed nodes.
+
+        A v1 yml file is laid out like this ('variables' written
+        bash-curly-brace style):
+
+            ${model_name}:
+                constraints:
+                    ${constraint_type}:
+                        - ${column_1}
+                        - ${column_2}
+                    ${other_constraint_type}:
+                        - ...
+            ${other_model_name}:
+                constraints:
+                    ...
+        """
+        for model_name, test_spec in test_yml.items():
+            # in v1 we can really only have constraints (right?), so not having
```

Comment:
that is correct! We added this because people (myself included) would write `contraints` and be confused when the tests didn't run :)